### PR TITLE
Added k8s-extension steps and decoupled installation

### PR DESCRIPTION
**Pull request by nshankar13**
Signed-off-by: nshankar13 <nshankar@microsoft.com>

**Description**: Remove "AZ Rest" and replace with "az k8s-extension" for install and update steps.

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Networking             [ ]
- Metrics                [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [X]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: scripts/install-k8s-extension.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
```
Comment (draychev) [inline scripts/install-k8s-extension.sh:2]: I'd highly recommend `set -aueo pipefail` here (in another PR or whenever)

And the `u` in that will complain about `$SUBSCRIPTION` (and a few of the other variables) most likely 🤔

File: scripts/install-connectedk8s.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
```
Comment (draychev) [inline scripts/install-connectedk8s.sh:2]: We could add `set -aueo pipefail` here too

File: scripts/install-connectedk8s.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+CONNECTEDK8S_VERSION="${CONNECTEDK8S_VERSION:-0.3.5}"
+
+az account set --subscription="$SUBSCRIPTION" > /dev/null 2>&1
```
Comment (draychev) [inline scripts/install-connectedk8s.sh:5]: If you don't want to use `az account set ...` you can pass `--subscription=...` to all the other `az` CLI commands - just FYI

**Review by draychev — APPROVED** [review]
Thank you @nshankar13 !
LGTM

**Review by SanyaKochhar — APPROVED** [review]
Love this change!

**Pull request merged**